src/main/java/com/project/videotecha/service/impl/OmdbServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.project.videotecha.service.impl;
+
+import com.project.videotecha.dto.MovieImdbDetailsDto;
+import com.project.videotecha.dto.OmdbSearchResultDto;
+import com.project.videotecha.exception.BusinessRuleException;
+import com.project.videotecha.service.OmdbService;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class OmdbServiceImpl implements OmdbService {
+
+    @Value("${omdbAPI.search-by-title}")
+    private String urlTitle;
+
+    @Value("${omdbAPI.search-by-imdbID}")
+    private String urlImdbID;
+
+    private final RestTemplate restTemplate;
+
+    public OmdbServiceImpl(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    @Override
+    public OmdbSearchResultDto searchMoviesFromOmdb(String title) {
+        try {
+            ResponseEntity<OmdbSearchResultDto> searchResult = restTemplate.getForEntity(urlTitle + title, OmdbSearchResultDto.class);
+            return searchResult.getBody();
+        } catch (Exception e) {
+            throw new BusinessRuleException("Error getting movies from OMDB API");
+        }
+    }
+
+    @Override
+    public MovieImdbDetailsDto getMovieDetailsFromImdb(String imdbId) {
```

Comment:
I'm confused about the naming, you're using OMDb api but in a few places in code you use `imdbId`. Why?